scripts/build/push-to-pages.sh

```diff
@@ -1,7 +1,6 @@
 git -C $GITHUB_PAGES_DIR config user.email "travis@users.noreply.github.com"
 git -C $GITHUB_PAGES_DIR config user.name travis
-# git -C $GITHUB_PAGES_DIR config credential.helper 'store --file ~/.travis.git.credentials'
-# echo https://$GIT_USERNAME:$GIT_PASSWORD@github.com >> ~/.travis.git.credentials
+git -C $GITHUB_PAGES_DIR config core.sshCommand 'ssh -i github_deploy_key'
```

Comment:
Could you please explain a bit more in the commit message on "why?" of this commit? I don't really understand what's going on here, what was bad in the previous code and why the new one is better :/ (also, similar for the change in build-chart-repo.sh)

Reply:
I updated commit message.